gammapy/modeling/models/temporal.py

```diff
@@ -657,6 +679,53 @@ def to_dict(self, full_output=False, format="table"):
         data["temporal"]["unit"] = str(self.map.unit)
         return data
 
+    def plot(self, time_range, energy=None, ax=None, n_points=100, **kwargs):
+        """
+        Plot Temporal Model.
+
+        Parameters
+        ----------
+        time_range : `~astropy.time.Time`
+            times to plot the model
+        energy : `~astropy.units.quantity`
+            energies to compute the model at for energy dependent models, optional
+        ax : `~matplotlib.axes.Axes`, optional
+            Axis to plot on
+        n_points : int
+            Number of bins to plot model
+        **kwargs : dict
+            Keywords forwarded to `~matplotlib.pyplot.errorbar`
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`, optional
+            axis
+        """
+        if not self.is_energy_dependent:
+            super().plot(time_range=time_range, ax=ax, n_points=n_points, **kwargs)
+        else:
+            time_min, time_max = time_range
+            time_axis = TimeMapAxis.from_time_bounds(
+                time_min=time_min, time_max=time_max, nbin=n_points
+            )
+            if energy is None:
+                energy_axis = self.map.geom.axes["energy"]
+            else:
+                energy_axis = MapAxis.from_nodes(
+                    nodes=energy, name="energy", interp="log"
+                )
+
+            m = RegionNDMap.create(region=None, axes=[time_axis, energy_axis])
+            kwargs.setdefault("marker", "None")
+            kwargs.setdefault("ls", "-")
+            m.quantity = self.evaluate(
+                time=time_axis.time_mid, energy=energy_axis.center
+            )
+            ax = m.plot(axis_name="time", ax=ax, **kwargs)
+            ax.set_ylabel("Norm / A.U.")
```

Comment:
shouldn't this be in flux units if the model is energy dependent?

Reply:
Well, we still take the unit from the spectral model, no?